src/sqs/sqs.ts

```diff
@@ -270,9 +301,11 @@ export class SqsQueue<MessagePayload extends BaseMessagePayload> {
 
 		try {
 			const messageProcessingStartTime = Date.now();
-
 			const rateLimitCheckResult = await preemptiveRateLimitCheck(context, this);
 			if (rateLimitCheckResult.isExceedThreshold) {
+
+				// Remove stale messages
+				if (await this.deleteStaleMessages(message, context)) return;
```

Comment:
I was wondering if we should be more aggressive and move it outside of rateLimitCheck maybe?